buffer.go

```diff
@@ -269,10 +273,17 @@ func (buf *Buffer) configure(schema *Schema) {
 			if sortingColumns[sortingIndex].Descending() {
 				column = &reversedColumnBuffer{column}
 			}
-			buf.sorted[sortingIndex] = column
+			sortedPlaceholders[sortingIndex] = column
 		}
 	})
 
+	buf.sorted = make([]ColumnBuffer, 0, len(sortingColumns))
+	for _, placeholder := range sortedPlaceholders {
+		if placeholder != nil {
+			buf.sorted = append(buf.sorted, placeholder)
+		}
+	}
+
```

Comment:
I think this whole thing can be simplified, we don't need the intermediary sortedPlaceholders slice, and you can use slices.DeleteFunc to remove the nil entries.

Reply:
Thanks for this @achille-roussel. Yeah, `slices.DeleteFunc` works. I did not see that the project uses Go 1.22, so yeah.